async-nats/src/subject.rs

```diff
@@ -105,6 +105,14 @@ impl From<String> for Subject {
     }
 }
 
+impl FromStr for Subject {
```

Comment:
There is no need for this trait as we have

```rust
impl From<String> for Subject {
    fn from(s: String) -> Self {
        // Since the input `String` is guaranteed to be valid UTF-8, we can
        // safely transmute the internal Vec<u8> to a Bytes value.
        let bytes = Bytes::from(s.into_bytes());
        Subject { bytes }
    }
}

```

Reply:
@Jarema I included this to round out the stdlib traits - where I'm using a `tokio_util::codec::Framed` to parse message frames, having an `FromStr` impl makes it easier to write utility parsing functions (rather than resort to `TryFrom<&str>`.

Reply:
From<String> and FromStr are not the same. FromStr is from a ro slice of memory and From<String> is from taking ownership of a owned copy. Very different usecases

Reply:
I didn't claim they were the same, only that one (`FromStr`) can be implemented by the other (`From`) and the former is useful in contexts where the latter isn't.